codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,217 @@ public TextHeaders clear() {
+
+        private final ValueConverter<CharSequence> valueConverter = valueConverter();
+        private final CsvValueQuoter<Object> objectQuoter = new CsvValueQuoter<Object>() {
+            @Override
+            public CharSequence quote(Object value) {
+                return StringUtil.quoteCsv(valueConverter.convertObject(value));
+            }
+        };
+        private final CsvValueQuoter<CharSequence> charSequenceQuoter = new CsvValueQuoter<CharSequence>() {
+            @Override
+            public CharSequence quote(CharSequence value) {
+                return StringUtil.quoteCsv(value);
+            }
+        };
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence value) {
+            return addQuotedValue(name, StringUtil.quoteCsv(value));
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            return addQuotedValue(name, commaSeparate(charSequenceQuoter, values));
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            return addQuotedValue(name, commaSeparate(charSequenceQuoter, values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            return addQuotedValue(name, commaSeparate(objectQuoter, values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            return addQuotedValue(name, commaSeparate(objectQuoter, values));
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        private TextHeaders addQuotedValue(CharSequence name, CharSequence escapedValue) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, escapedValue);
+            } else {
+                if (!currentValue.equals(escapedValue)) {
```

Comment:
Is this correct?  Consider the following example:

```java
headers.add("foo", "goo");
headers.add("foo", "goo");
```

Shouldn't the result be `foo: goo, goo`?

It seems like we should not be filtering additions based upon the value.  Should we add a unit test for this?